src/shared/modules/dbMeta/dbMetaDuck.ts

```diff
@@ -699,5 +710,14 @@ export const serverInfoEpic = (some$: any, store: any) =>
     })
     .mapTo({ type: 'NOOP' })
 
-export const clearMetaOnDisconnectEpic = (some$: any, _store: any) =>
-  some$.ofType(DISCONNECTION_SUCCESS).mapTo({ type: CLEAR })
+export const clearMetaOnDisconnectEpic = (some$: any, store: any) =>
+  some$
+    .ofType(DISCONNECTION_SUCCESS)
+    .merge(some$.ofType(SWITCH_CONNECTION))
+    .merge(some$.ofType(SILENT_DISCONNECT))
```

Comment:
This already happens in https://github.com/neo4j/neo4j-browser/blob/f76ff9c731f62ecb606d715872d6aed96f273885/src/shared/modules/connections/connectionsDuck.ts#L581-L588 so maybe create a separate epic just to handle this case so we don't get any unexpected side effects.